packages/devtools_app/integration_test/test/live_connection/service_connection_test.dart

```diff
@@ -24,64 +29,206 @@ void main() {
-      const Range(50, 60).contains(vmServiceCallCount),
-      isTrue,
-      reason: 'Unexpected number of vm service calls upon connection: '
-          '$vmServiceCallCount. If this is expected, please update this test '
-          'to the new expected number of calls. Here are the calls for this '
-          'test run:\n ${serviceManager.service!.vmServiceCalls.toString()}',
+    logStatus('verify toggling a boolean service extension');
+    final extensionName = extensions.debugPaint.extension;
+    const evalExpression = 'debugPaintSizeEnabled';
+    final library = EvalOnDartLibrary(
+      'package:flutter/src/rendering/debug.dart',
+      serviceManager.service!,
     );
-    // Check the ordering of the vm service calls we can expect to occur
-    // in a stable order.
-    expect(
-      serviceManager.service!.vmServiceCalls
-          // Filter out unawaited streamListen calls.
-          .where((call) => call != 'streamListen')
-          .toList()
-          .sublist(0, 5),
-      equals([
-        'getSupportedProtocols',
-        'getVersion',
-        'getFlagList',
-        'getVM',
-        'getIsolate',
-      ]),
+
+    await _serviceExtensionAvailable(extensionName);
+
+    logStatus('verify initial state on device is false');
+
+    // This chunk of code is behaving weirdly
+    logStatus('before eval $evalExpression');
+    final result = await library.eval(evalExpression, isAlive: null);
+    logStatus('after eval $evalExpression');
+    if (result is InstanceRef) {
+      logStatus(
+        'result.valueAsString: ${result.valueAsString}, expectedResult: false',
+      );
+      if (result.valueAsString == 'false') {
+        logStatus('this expectation is true');
+      }
+      // This expectation is failing after the test finishes, even though it
+      // does not fail during the test execution.
+      logStatus(
+          'before calling "expect(result.valueAsString, equals(\'false\'))"');
+      expect(result.valueAsString, equals('false'));
+      logStatus(
+          'after calling "expect(result.valueAsString, equals(\'false\'))"');
+    }
+    // end chunk
+
+    // await _verifyExtensionStateOnTestDevice(
+    //   evalExpression: evalExpression,
+    //   expectedResult: 'false',
+    //   library: library,
+    // );
+
+    logStatus('verify initial state in service manager is false');
+    await _verifyInitialExtensionStateInServiceManager(extensionName);
+
+    // The test only fails when the following block is present, but oddly it
+    // fails above at line 127 `expect(result.valueAsString, equals('false'));`
+    // however it fails at line 127 after the test has already completed and the
+    // prints show we have already successfully made it past line 127.
+    logStatus('enable the service extension via ServiceExtensionManager');
+    await serviceManager.serviceExtensionManager.setServiceExtensionState(
+      extensionName,
+      enabled: true,
+      value: true,
```

Comment:
With this line added, the test fails with this debug output:
```
FlutterDriveProcess - Launching integration_test/test/live_connection/service_connection_test.dart on Chrome in debug mode...
FlutterDriveProcess - Waiting for connection from debug service on Chrome...             47.2s
FlutterDriveProcess - This app is linked to the debug service: ws://127.0.0.1:58730/xN9wiVXKPcE=/ws
FlutterDriveProcess - Debug service listening on ws://127.0.0.1:58730/xN9wiVXKPcE=/ws
FlutterDriveProcess - The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
FlutterDriveProcess - 00:00 +0: (setUpAll)
FlutterDriveProcess - 00:00 +1: can call service extensions
FlutterDriveProcess - devtools server not available (200)
FlutterDriveProcess - DevTools version 2.25.0-dev.5.
FlutterDriveProcess - XXXXX WebDriver.connectWeb
FlutterDriveProcess - XXXXX WebDriver.connectedTo
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: Instance of 'RequestData'
FlutterDriveProcess - XXXXX before _connection.sendCommand serialized: {command: request_data, message: pending}
FlutterDriveProcess - TEST STATUS: verify that we can connect to an app
FlutterDriveProcess - Attempted to call extension 'ext.flutter.inspector.structuredErrors', but no service with that name exists
FlutterDriveProcess - TEST STATUS: verify toggling a boolean service extension
FlutterDriveProcess - TEST STATUS: verify initial state on device is false
FlutterDriveProcess - TEST STATUS: before eval debugPaintSizeEnabled
FlutterDriveProcess - TEST STATUS: after eval debugPaintSizeEnabled
FlutterDriveProcess - TEST STATUS: result.valueAsString: false, expectedResult: false
FlutterDriveProcess - TEST STATUS: this expectation is true
FlutterDriveProcess - TEST STATUS: before calling "expect(result.valueAsString, equals('false'))"
FlutterDriveProcess - TEST STATUS: after calling "expect(result.valueAsString, equals('false'))"
FlutterDriveProcess - TEST STATUS: verify initial state in service manager is false
FlutterDriveProcess - TEST STATUS: _verifyExtensionStateInServiceManager, ext.flutter.debugPaint - enabled: false, value: null
FlutterDriveProcess - in stateListener - null
FlutterDriveProcess - TEST STATUS: before await future in verify in servicemanager fx
FlutterDriveProcess - TEST STATUS: after await future in verify in servicemanager fx
FlutterDriveProcess - TEST STATUS: enable the service extension via ServiceExtensionManager
FlutterDriveProcess - calling the service extension ext.flutter.debugPaint
FlutterDriveProcess - TEST STATUS: at the end of the test
FlutterDriveProcess - XXXXXX runTest - setting results: key = can call service extensions, value = success
FlutterDriveProcess - 00:36 +2: (tearDownAll)
FlutterDriveProcess - XXXXXX completing all tests passed from tearDownAll. failureMethodsDetails:
FlutterDriveProcess -  []
FlutterDriveProcess - XXXXXX _callback_web cleanup
FlutterDriveProcess - XXXXXX _callback_web cleanup sending noop
FlutterDriveProcess - XXXXXX _callback_web cleanup completeing driver command
FlutterDriveProcess - XXXXXX returning because kIsWeb true
FlutterDriveProcess - 00:36 +3: All tests passed!
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: data = {"isError":false,"response":{"message":"{\"result\":\"false\",\"failureDetails\":[],\"data\":{\"web_driver_command\":\"WebDriverCommandType.noop\"}}"}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: returning {message: {"result":"false","failureDetails":[],"data":{"web_driver_command":"WebDriverCommandType.noop"}}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: Instance of 'RequestData'
FlutterDriveProcess - XXXXX before _connection.sendCommand serialized: {command: request_data}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: data = {"isError":false,"response":{"message":"{\"result\":\"false\",\"failureDetails\":[\"{\\\"methodName\\\":\\\"can call service extensions\\\",\\\"details\\\":\\\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\\\nThe following TestFailure was thrown running a test:\\\\nExpected: 'false'\\\\n  Actual: 'true'\\\\n   Which: is different.\\\\n          Expected: false\\\\n            Actual: true\\\\n                    ^\\\\n           Differ at offset 0\\\\n\\\\nWhen the exception was thrown, this was the stack:\\\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\\\nservice_connection_test.dart 127:7                                            <fn>\\\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\\\n\\\\nThe test description was:\\\\n  can call service extensions\\\\n═════════════════════════════════════════════════════════════════\\\\n\\\"}\"]}"}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: returning {message: {"result":"false","failureDetails":["{\"methodName\":\"can call service extensions\",\"details\":\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\nThe following TestFailure was thrown running a test:\\nExpected: 'false'\\n  Actual: 'true'\\n   Which: is different.\\n          Expected: false\\n            Actual: true\\n                    ^\\n           Differ at offset 0\\n\\nWhen the exception was thrown, this was the stack:\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\nservice_connection_test.dart 127:7                                            <fn>\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\n\\nThe test description was:\\n  can call service extensions\\n═════════════════════════════════════════════════════════════════\\n\"}"]}}
FlutterDriveProcess - XXXXXX If No-op command is sent, ask for the result of all tests
FlutterDriveProcess - result {"result":"false","failureDetails":["{\"methodName\":\"can call service extensions\",\"details\":\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\nThe following TestFailure was thrown running a test:\\nExpected: 'false'\\n  Actual: 'true'\\n   Which: is different.\\n          Expected: false\\n            Actual: true\\n                    ^\\n           Differ at offset 0\\n\\nWhen the exception was thrown, this was the stack:\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\nservice_connection_test.dart 127:7                                            <fn>\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\n\\nThe test description was:\\n  can call service extensions\\n═════════════════════════════════════════════════════════════════\\n\"}"]}
FlutterDriveProcess - Failure Details:
FlutterDriveProcess - Failure in method: can call service extensions
FlutterDriveProcess - ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
FlutterDriveProcess - The following TestFailure was thrown running a test:
FlutterDriveProcess - Expected: 'false'
FlutterDriveProcess -   Actual: 'true'
FlutterDriveProcess -    Which: is different.
FlutterDriveProcess -           Expected: false
FlutterDriveProcess -             Actual: true
FlutterDriveProcess -                     ^
FlutterDriveProcess -            Differ at offset 0
FlutterDriveProcess - 
FlutterDriveProcess - When the exception was thrown, this was the stack:
FlutterDriveProcess - dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_
FlutterDriveProcess - packages/matcher/src/expect/expect.dart 149:31                                fail
FlutterDriveProcess - packages/matcher/src/expect/expect.dart 144:3                                 _expect
FlutterDriveProcess - packages/matcher/src/expect/expect.dart 56:3                                  expect$
FlutterDriveProcess - packages/flutter_test/src/widget_tester.dart 467:18                           expect$
FlutterDriveProcess - service_connection_test.dart 127:7                                            <fn>
FlutterDriveProcess - dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1308:19                                          runUnary
FlutterDriveProcess - dart-sdk/lib/async/future_impl.dart 156:18                                    handleValue
FlutterDriveProcess - dart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback
FlutterDriveProcess - dart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners
FlutterDriveProcess - dart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]
FlutterDriveProcess - dart-sdk/lib/async/future_impl.dart 715:7                                     <fn>
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1399:13                                          _rootRun
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1301:19                                          run
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1209:7                                           runGuarded
FlutterDriveProcess - dart-sdk/lib/async/zone.dart 1249:23                                          callback
FlutterDriveProcess - dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
FlutterDriveProcess - dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
FlutterDriveProcess - dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>
FlutterDriveProcess - 
FlutterDriveProcess - The test description was:
FlutterDriveProcess -   can call service extensions
FlutterDriveProcess - ═════════════════════════════════════════════════════════════════
FlutterDriveProcess - 
FlutterDriveProcess - end of failure 1
FlutterDriveProcess - 
FlutterDriveProcess - 
FlutterDriveProcess - 
FlutterDriveProcess - Application finished.
flutter drive process has exited
throwing exception because buffer has content
killing the test app
cancelling stream subscriptions
killing the chromedriver process
Unhandled exception:
Exception: XXXXXX WebDriver.sendCommand: data = {"isError":false,"response":{"message":"{\"result\":\"false\",\"failureDetails\":[\"{\\\"methodName\\\":\\\"can call service extensions\\\",\\\"details\\\":\\\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\\\nThe following TestFailure was thrown running a test:\\\\nExpected: 'false'\\\\n  Actual: 'true'\\\\n   Which: is different.\\\\n          Expected: false\\\\n            Actual: true\\\\n                    ^\\\\n           Differ at offset 0\\\\n\\\\nWhen the exception was thrown, this was the stack:\\\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\\\nservice_connection_test.dart 127:7                                            <fn>\\\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\\\n\\\\nThe test description was:\\\\n  can call service extensions\\\\n═════════════════════════════════════════════════════════════════\\\\n\\\"}\"]}"}}
XXXXXX WebDriver.sendCommand: returning {message: {"result":"false","failureDetails":["{\"methodName\":\"can call service extensions\",\"details\":\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\nThe following TestFailure was thrown running a test:\\nExpected: 'false'\\n  Actual: 'true'\\n   Which: is different.\\n          Expected: false\\n            Actual: true\\n                    ^\\n           Differ at offset 0\\n\\nWhen the exception was thrown, this was the stack:\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\nservice_connection_test.dart 127:7                                            <fn>\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\n\\nThe test description was:\\n  can call service extensions\\n═════════════════════════════════════════════════════════════════\\n\"}"]}}
XXXXXX If No-op command is sent, ask for the result of all tests
Test 'can call service extensions' failed: ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
The following TestFailure was thrown running a test:
Expected: 'false'
  Actual: 'true'
   Which: is different.
          Expected: false
            Actual: true
                    ^
           Differ at offset 0

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_
packages/matcher/src/expect/expect.dart 149:31                                fail
packages/matcher/src/expect/expect.dart 144:3                                 _expect
packages/matcher/src/expect/expect.dart 56:3                                  expect$
packages/flutter_test/src/widget_tester.dart 467:18                           expect$
service_connection_test.dart 127:7                                            <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
dart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary
dart-sdk/lib/async/zone.dart 1308:19                                          runUnary
dart-sdk/lib/async/future_impl.dart 156:18                                    handleValue
dart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback
dart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners
dart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]
dart-sdk/lib/async/future_impl.dart 715:7                                     <fn>
dart-sdk/lib/async/zone.dart 1399:13                                          _rootRun
dart-sdk/lib/async/zone.dart 1301:19                                          run
dart-sdk/lib/async/zone.dart 1209:7                                           runGuarded
dart-sdk/lib/async/zone.dart 1249:23                                          callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>

The test description was:
  can call service extensions
═════════════════════════════════════════════════════════════════
.

result {"result":"false","failureDetails":["{\"methodName\":\"can call service extensions\",\"details\":\"══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════\\nThe following TestFailure was thrown running a test:\\nExpected: 'false'\\n  Actual: 'true'\\n   Which: is different.\\n          Expected: false\\n            Actual: true\\n                    ^\\n           Differ at offset 0\\n\\nWhen the exception was thrown, this was the stack:\\ndart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_\\npackages/matcher/src/expect/expect.dart 149:31                                fail\\npackages/matcher/src/expect/expect.dart 144:3                                 _expect\\npackages/matcher/src/expect/expect.dart 56:3                                  expect$\\npackages/flutter_test/src/widget_tester.dart 467:18                           expect$\\nservice_connection_test.dart 127:7                                            <fn>\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>\\ndart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary\\ndart-sdk/lib/async/zone.dart 1308:19                                          runUnary\\ndart-sdk/lib/async/future_impl.dart 156:18                                    handleValue\\ndart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback\\ndart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners\\ndart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]\\ndart-sdk/lib/async/future_impl.dart 715:7                                     <fn>\\ndart-sdk/lib/async/zone.dart 1399:13                                          _rootRun\\ndart-sdk/lib/async/zone.dart 1301:19                                          run\\ndart-sdk/lib/async/zone.dart 1209:7                                           runGuarded\\ndart-sdk/lib/async/zone.dart 1249:23                                          callback\\ndart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop\\ndart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop\\ndart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>\\n\\nThe test description was:\\n  can call service extensions\\n═════════════════════════════════════════════════════════════════\\n\"}"]}
Failure Details:
Failure in method: can call service extensions
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
The following TestFailure was thrown running a test:
Expected: 'false'
  Actual: 'true'
   Which: is different.
          Expected: false
            Actual: true
                    ^
           Differ at offset 0

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49  throw_
packages/matcher/src/expect/expect.dart 149:31                                fail
packages/matcher/src/expect/expect.dart 144:3                                 _expect
packages/matcher/src/expect/expect.dart 56:3                                  expect$
packages/flutter_test/src/widget_tester.dart 467:18                           expect$
service_connection_test.dart 127:7                                            <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
dart-sdk/lib/async/zone.dart 1407:47                                          _rootRunUnary
dart-sdk/lib/async/zone.dart 1308:19                                          runUnary
dart-sdk/lib/async/future_impl.dart 156:18                                    handleValue
dart-sdk/lib/async/future_impl.dart 840:44                                    handleValueCallback
dart-sdk/lib/async/future_impl.dart 869:13                                    _propagateToListeners
dart-sdk/lib/async/future_impl.dart 641:5                                     [_completeWithValue]
dart-sdk/lib/async/future_impl.dart 715:7                                     <fn>
dart-sdk/lib/async/zone.dart 1399:13                                          _rootRun
dart-sdk/lib/async/zone.dart 1301:19                                          run
dart-sdk/lib/async/zone.dart 1209:7                                           runGuarded
dart-sdk/lib/async/zone.dart 1249:23                                          callback
dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:15           <fn>

The test description was:
  can call service extensions
═════════════════════════════════════════════════════════════════


#0      runFlutterIntegrationTest (file:///Users/kenzieschmoll/develop/devtools/packages/devtools_app/integration_test/test_infra/run/run_test.dart:88:5)
<asynchronous suspension>
#1      _runTest (file:///Users/kenzieschmoll/develop/devtools/packages/devtools_app/integration_test/run_tests.dart:64:3)
<asynchronous suspension>
#2      main (file:///Users/kenzieschmoll/develop/devtools/packages/devtools_app/integration_test/run_tests.dart:39:5)
<asynchronous suspension>
```

Reply:
Without the call to `setServiceExtensionState`
```
FlutterDriveProcess - Launching integration_test/test/live_connection/service_connection_test.dart on Chrome in debug mode...
FlutterDriveProcess - Waiting for connection from debug service on Chrome...             45.8s
FlutterDriveProcess - This app is linked to the debug service: ws://127.0.0.1:59244/shxzNWCwtc0=/ws
FlutterDriveProcess - Debug service listening on ws://127.0.0.1:59244/shxzNWCwtc0=/ws
FlutterDriveProcess - The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
FlutterDriveProcess - 00:00 +0: (setUpAll)
FlutterDriveProcess - 00:00 +1: can call service extensions
FlutterDriveProcess - devtools server not available (200)
FlutterDriveProcess - DevTools version 2.25.0-dev.5.
FlutterDriveProcess - XXXXX WebDriver.connectWeb
FlutterDriveProcess - XXXXX WebDriver.connectedTo
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: Instance of 'RequestData'
FlutterDriveProcess - XXXXX before _connection.sendCommand serialized: {command: request_data, message: pending}
FlutterDriveProcess - TEST STATUS: verify that we can connect to an app
FlutterDriveProcess - Attempted to call extension 'ext.flutter.inspector.structuredErrors', but no service with that name exists
FlutterDriveProcess - TEST STATUS: verify toggling a boolean service extension
FlutterDriveProcess - TEST STATUS: verify initial state on device is false
FlutterDriveProcess - TEST STATUS: before eval debugPaintSizeEnabled
FlutterDriveProcess - TEST STATUS: after eval debugPaintSizeEnabled
FlutterDriveProcess - TEST STATUS: result.valueAsString: false, expectedResult: false
FlutterDriveProcess - TEST STATUS: this expectation is true
FlutterDriveProcess - TEST STATUS: before calling "expect(result.valueAsString, equals('false'))"
FlutterDriveProcess - TEST STATUS: after calling "expect(result.valueAsString, equals('false'))"
FlutterDriveProcess - TEST STATUS: verify initial state in service manager is false
FlutterDriveProcess - TEST STATUS: _verifyExtensionStateInServiceManager, ext.flutter.debugPaint - enabled: false, value: null
FlutterDriveProcess - in stateListener - null
FlutterDriveProcess - TEST STATUS: before await future in verify in servicemanager fx
FlutterDriveProcess - TEST STATUS: after await future in verify in servicemanager fx
FlutterDriveProcess - TEST STATUS: at the end of the test
FlutterDriveProcess - XXXXXX runTest - setting results: key = can call service extensions, value = success
FlutterDriveProcess - 00:36 +2: (tearDownAll)
FlutterDriveProcess - XXXXXX completing all tests passed from tearDownAll. failureMethodsDetails:
FlutterDriveProcess -  []
FlutterDriveProcess - XXXXXX _callback_web cleanup
FlutterDriveProcess - XXXXXX _callback_web cleanup sending noop
FlutterDriveProcess - XXXXXX _callback_web cleanup completeing driver command
FlutterDriveProcess - XXXXXX returning because kIsWeb true
FlutterDriveProcess - 00:37 +3: All tests passed!
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: data = {"isError":false,"response":{"message":"{\"result\":\"false\",\"failureDetails\":[],\"data\":{\"web_driver_command\":\"WebDriverCommandType.noop\"}}"}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: returning {message: {"result":"false","failureDetails":[],"data":{"web_driver_command":"WebDriverCommandType.noop"}}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: Instance of 'RequestData'
FlutterDriveProcess - XXXXX before _connection.sendCommand serialized: {command: request_data}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: data = {"isError":false,"response":{"message":"{\"result\":\"true\",\"failureDetails\":[]}"}}
FlutterDriveProcess - XXXXXX WebDriver.sendCommand: returning {message: {"result":"true","failureDetails":[]}}
FlutterDriveProcess - XXXXXX If No-op command is sent, ask for the result of all tests
FlutterDriveProcess - result {"result":"true","failureDetails":[]}
FlutterDriveProcess - All tests passed.
FlutterDriveProcess - Application finished.
flutter drive process has exited
killing the test app
cancelling stream subscriptions
killing the chromedriver process
```